neuro_san/session/async_http_service_agent_session.py

```diff
@@ -0,0 +1,107 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import Dict
+from typing import Generator
+
+import json
+
+from aiohttp import ClientSession
+
+from neuro_san.interfaces.async_agent_session import AsyncAgentSession
+from neuro_san.session.abstract_http_service_agent_session import AbstractHttpServiceAgentSession
+
+
+class AsyncHttpServiceAgentSession(AbstractHttpServiceAgentSession, AsyncAgentSession):
+    """
+    Implementation of AsyncAgentSession that talks to an HTTP service.
+    """
```

Comment:
New async implementation for HTTP access.
Worth noting that is uses aiohttp, which seems to be the closest cousin to synchronous requests library used in synchronous version.